src/DIRAC/Core/Utilities/Subprocess.py

```diff
@@ -241,60 +241,58 @@ def __selectFD(self, readSeq, timeout=False):
         events = sel.select(timeout=timeout or self.timeout or None)
         return [key.fileobj for key, event in events if event & selectors.EVENT_READ]
 
-    def __killPid(self, pid, sig=9):
-        """send signal :sig: to process :pid:
-
-        :param int pid: process id
-        :param int sig: signal to send, default 9 (SIGKILL)
-        """
+    def __killProcess(self, process):
+        """Kills a Process with SIGTERM and returns a (gone, alive) tuple"""
+        self.log.verbose(f"Sending kill signal to PID {process.pid}")
         try:
-            os.kill(pid, sig)
-        except Exception as x:
-            if str(x) != "[Errno 3] No such process":
-                self.log.exception("Exception while killing timed out process")
-                raise x
+            process.send_signal(signal.SIGTERM)
```

Comment:
You could potentially use `terminate()`: https://psutil.readthedocs.io/en/latest/index.html#psutil.Process.terminate
```suggestion
            process.terminate()
```